### PR TITLE
Add openstack provider

### PR DESCRIPTION
**Pull request by ericgraf**

### CODE REVIEW

**Review by whites11 — APPROVED** [review]
LGTMIIWFY

File: manifests/provider/openstack/override.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+kind: AppProject
+metadata:
+  name: collections
+  namespace: argocd
+spec:
+  description: Management Cluster openstack Collection
+  sourceRepos:
+    - 'https://github.com/giantswarm/config.git'
+    - 'https://github.com/giantswarm/openstack-app-collection.git'
+    - 'https://github.com/giantswarm/rest-api-app-collection.git'
```
Comment (tfussell) [inline manifests/provider/openstack/override.yaml:21]: Are we sure we want these legacy components running on the CAPO MCs?

File: manifests/provider/openstack/override.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+apiVersion: argoproj.io/v1alpha1
+kind: AppProject
+metadata:
+  name: collections
+  namespace: argocd
+spec:
+  description: Management Cluster openstack Collection
+  sourceRepos:
+    - 'https://github.com/giantswarm/config.git'
+    - 'https://github.com/giantswarm/openstack-app-collection.git'
```
Comment (MarcelMue) [inline manifests/provider/openstack/override.yaml:20]: Do we have all the required apps in here already? 
We should bring up the idea of a shared collection again, this is going to get more difficult with more providers.

File: manifests/provider/openstack/override.yaml (hunk 1 above)
Comment (MarcelMue) [reply manifests/provider/openstack/override.yaml:21]: I think some Happa functionality may still rely on it currently.

Comment (MarcelMue) [reply manifests/provider/openstack/override.yaml:21]: Maybe @giantswarm/team-rainbow can confirm?

Comment (anvddriesch) [reply manifests/provider/openstack/override.yaml:21]: AFAIK happa should be able to function entirely without these components if SSO is enabled.

Comment (ericgraf) [reply manifests/provider/openstack/override.yaml:21]: I will remove it for now and if required we can re-add it.

File: manifests/provider/openstack/override.yaml (hunk 2 above)
Comment (ericgraf) [reply manifests/provider/openstack/override.yaml:20]: Day 1 we have https://github.com/giantswarm/openstack-app-collection . Currently being worked on. Day 2 we should discuss a shared collection. @MarcelMue would you be okay to open a ticket to track that if it doesn't already exist?

**Comment by ericgraf** [comment]
rest-api-app-collection has been deleted for today.  If things fail we might need to re-add it.

**Review by kopiczko — APPROVED** [review]
LGTM

File: manifests/provider/openstack/override.yaml (hunk 2 above)
Comment (MarcelMue) [reply manifests/provider/openstack/override.yaml:20]: We had this here but it died:
https://github.com/giantswarm/giantswarm/issues/18741
Will open an issue after slack discussion

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**